subsys/net/l2/ethernet/ethernet.c

```diff
@@ -116,10 +116,10 @@ static inline void ethernet_update_length(struct net_if *iface,
 	}
 }
 
-#if defined(CONFIG_NET_STATISTICS_ETHERNET)
 static void ethernet_update_rx_stats(struct net_if *iface,
 				     struct net_pkt *pkt, size_t length)
 {
+#if defined(CONFIG_NET_STATISTICS_ETHERNET)
```

Comment:
The issue with this is that some compiler options will generate warnings on unused arguments. I don't see why you change that

Reply:
I removed ifdefs around updating the error stat.

Take another look at the sucessful case, there's no need for dummy here. Or, am I missing anything?

Reply:
I mean all #ifdef about CONFIG_NET_STATISTICS_ETHERNET can be removed here, since eth_stats_update_bytes_rx(), eth_stats_update_pkts_rx()  and so on are already provided as dummies in eth_stats.h
So compiler will optimize by itself when it will see empty statements in the if/else: it will remove this part entirely. (in case CONFIG_NET_STATISTICS_ETHERNET is unset)